simple_parsing/helpers/serialization/serializable.py

```diff
@@ -652,13 +662,23 @@ def to_dict(dc, dict_factory: type[dict] = dict, recurse: bool = True) -> dict:
             d[name] = custom_encoding_fn(value)
             continue
 
+        if add_selection and "subgroups" in f.metadata:
+            ###### insert subgroups selected key
+            subgroups_dict = f.metadata["subgroups"]
+            for g_name, g_cls in subgroups_dict.items():
+                if isinstance(value, g_cls):
```

Comment:
This check isn't exactly accurate: `subgroups_dict` can contain some `functools.partial` objects as values. What you'd want to use here is `f.metadata["subgroup_dataclass_types"]` : https://github.com/lebrice/SimpleParsing/blob/97ebedb607578f539b5f107120bbd6f479ad8a6f/simple_parsing/helpers/subgroups.py#L100-L102